From 064abd9b857d7440b7e2b98731cb88a322b07040 Mon Sep 17 00:00:00 2001
From: Electric Worry <electricworry@users.noreply.github.com>
Date: Sun, 12 Nov 2023 18:57:48 +0000
Subject: [PATCH 03/19] allwinner,sun50i-h616 cpufreq blocklist.

---
 drivers/cpufreq/cpufreq-dt-platdev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/cpufreq/cpufreq-dt-platdev.c b/drivers/cpufreq/cpufreq-dt-platdev.c
index 69a8742c0..f20d412ef 100644
--- a/drivers/cpufreq/cpufreq-dt-platdev.c
+++ b/drivers/cpufreq/cpufreq-dt-platdev.c
@@ -102,6 +102,7 @@ static const struct of_device_id allowlist[] __initconst = {
  */
 static const struct of_device_id blocklist[] __initconst = {
 	{ .compatible = "allwinner,sun50i-h6", },
+	{ .compatible = "allwinner,sun50i-h616", },
 
 	{ .compatible = "arm,vexpress", },
 
-- 
2.34.1

