From e9c7c63c9706b96333c87f13e3433e228ad72b90 Mon Sep 17 00:00:00 2001
From: Electric Worry <electricworry@users.noreply.github.com>
Date: Thu, 16 Nov 2023 12:46:57 +0000
Subject: [PATCH 13/19] allwinner,sun50i-h616-lradc

---
 drivers/input/keyboard/sun4i-lradc-keys.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/input/keyboard/sun4i-lradc-keys.c b/drivers/input/keyboard/sun4i-lradc-keys.c
index 15c15c095..ba952642f 100644
--- a/drivers/input/keyboard/sun4i-lradc-keys.c
+++ b/drivers/input/keyboard/sun4i-lradc-keys.c
@@ -86,6 +86,12 @@ static const struct lradc_variant lradc_variant_r329 = {
 	.has_clock_reset = true,
 };
 
+static const struct lradc_variant lradc_variant_h616 = {
+	.divisor_numerator = 3,
+	.divisor_denominator = 4,
+	.has_clock_reset = true,
+};
+
 struct sun4i_lradc_keymap {
 	u32 voltage;
 	u32 keycode;
@@ -345,6 +351,8 @@ static const struct of_device_id sun4i_lradc_of_match[] = {
 		.data = &r_lradc_variant_a83t },
 	{ .compatible = "allwinner,sun50i-r329-lradc",
 		.data = &lradc_variant_r329 },
+	{ .compatible = "allwinner,sun50i-h616-lradc",
+		.data = &lradc_variant_h616 },
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, sun4i_lradc_of_match);
-- 
2.34.1

